packages/contract/src/builder.ts

```diff
@@ -751,6 +751,35 @@ export type PublisherFirstResult<
   };
 };
 
+/**
+ * Publisher-first builder result for topic exchanges.
+ *
+ * This type represents a publisher with a concrete routing key and provides a method
+ * to create consumers that can use routing key patterns matching the publisher's key.
+ *
+ * @template TMessage - The message definition
+ * @template TPublisher - The publisher definition
+ */
+export type PublisherFirstResultWithRoutingKey<
+  TMessage extends MessageDefinition,
+  TPublisher extends PublisherDefinition<TMessage>,
+> = {
+  /** The publisher definition */
+  publisher: TPublisher;
+  /**
+   * Create a consumer that receives messages from this publisher.
+   * For topic exchanges, the routing key pattern can be specified for the binding.
+   *
+   * @param queue - The queue that will consume the messages
+   * @param routingKey - Optional routing key pattern for the binding (defaults to publisher's routing key)
+   * @returns An object with the consumer definition and binding
+   */
+  createConsumer: (queue: QueueDefinition, routingKey?: string) => {
```

Comment:
create a litteral type alias for routingKey, this type must derive from publisher routing key